### DISCUSSION TITLE
PG-Vector usage in apps that run on both Babelfish and MSSQL

### DISCUSSION THREAD

**Discussion by staticlibs** [babelfish-for-postgresql/babelfish-for-postgresql#14] [General]
I have a general question about the [PG-Vector](https://github.com/babelfish-for-postgresql/babelfish_extensions/pull/2309) feature.

Situation is the following: existing MSSQL application is being modified to support Babelfish. Assuming the Babelfish migration will actually happen in a reasonable time, the intention is to move all AWS deployments of this app from MSSQL to Aurora Babelfish. Though MSSQL remains being supported for on-premises customers that have their internal MSSQL-based infra and cannot be moved to Babelfish (neither standalone nor cloud).

Vector support in Babelfish caught and eye of some people, because it allows to implement a number of "down to earth" app features (like a special kind of search on vectors) that can be marketed as "based on AI technology". And to do this without introducing new kind of storage engines/services/dependencies to the app.

I am being asked "How are we going to run such app features on MSSQL?" and, after some research, I don't have an answer on this. Unlike all other Babelfish features, there seems to be no direct counterpart to PG-Vectors in MSSQL.

I wonder if there are any recommendations or opinions what can be done in this case?

**Comment by suprio-amzn** [comment]
Somethings may be happening there: see https://devblogs.microsoft.com/azure-sql/announcing-eap-native-vector-support-in-azure-sql-database/

**Comment by staticlibs** [comment]
This new `VECTOR_DISTANCE` seems to be an improvement over the previous examples ([1](https://devblogs.microsoft.com/azure-sql/vector-similarity-search-with-azure-sql-database-and-openai/), [2](https://devblogs.microsoft.com/azure-sql/vector-search-with-azure-sql-database/)). But it is Azure SQL only and no indication when this will be added to non-cloud versions of MSSQL.
